vortex-serde/src/layouts/read/batch.rs

```diff
@@ -53,14 +96,23 @@ impl BatchReader {
             let child_arrays = mem::replace(&mut self.arrays, vec![None; self.children.len()])
                 .into_iter()
                 .enumerate()
-                .map(|(i, a)| a.ok_or_else(|| vortex_err!("Missing child array at index {}", i)))
+                .map(|(i, a)| a.ok_or_else(|| vortex_err!("Missing child array at index {i}")))
                 .collect::<VortexResult<Vec<_>>>()?;
-            Ok(Some(ReadResult::Batch(
-                StructArray::from_fields(&self.names.iter().zip(child_arrays).collect::<Vec<_>>())?
-                    .into_array(),
-            )))
+            let len = child_arrays
```

Comment:
You need to know your own length. I know it's dumb, but this requires at least one child which isn't always true of struct arrays.

Reply:
Our own length in this case is the length of the selection we have been asked for